### PR TITLE
Fix raise error parsing

### LINKED ISSUE #208: Parse body only if response header specify it
I'm using a custom APEX REST endpoint and when this endpoint validation fails, it returns a 500 and the body is a simple string.

It looks like Restforce is not checking the `Content-Type` header before parsing the body:

https://github.com/ejholmes/restforce/blob/66ef4acdd0517e7b31a4105af242c1948723157f/lib/restforce/middleware/raise_error.rb#L23

### PR DESCRIPTION
**Pull request by kouno**
This fixes #208. This branch is on top of `kouno:fix-cache-read` to see all tests pass.

### CODE REVIEW

File: lib/restforce/middleware/caching.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ def cache_key(env)
     end
 
     def use_cache?
-      !@options.key?(:use_cache) || @options[:use_cache]
+      @options.fetch(:use_cache, true)
```
Comment (timrogers) [inline lib/restforce/middleware/caching.rb]: Good move here - if I'd written this I'd have use `#fetch` but didn't fix these when I did my refactor.

**Comment by timrogers** [comment]
:+1:

**Comment by timrogers** [comment]
I'll get this and #210 pushed as a new version ASAP.

**Comment by kouno** [comment]
Thanks @timrogers!

File: lib/restforce/middleware/raise_error.rb [hunk 2]
```diff
@@ -16,11 +16,18 @@ def on_complete(env)
     end
 
     def message
-      "#{body.first['errorCode']}: #{body.first['message']}"
+      "#{body['errorCode']}: #{body['message']}"
     end
 
     def body
-      JSON.parse(@env[:body])
+      @body = (@env[:body].respond_to?(:first) ? @env[:body].first : @env[:body])
```
Comment (kouno) [inline lib/restforce/middleware/raise_error.rb]: This didn't entirely fix my issue as ActiveRecord actually monkey patch String to add the `first` method. :cry:

I'll add a PR to fix this again.

**Pull request merged**